lidarr/trackfile.go

```diff
@@ -0,0 +1,230 @@
+	TotalHours        int64 `json:"totalHours"`
+	TotalMilliseconds int64 `json:"totalMilliseconds"`
+	TotalMinutes      int64 `json:"totalMinutes"`
+	TotalSeconds      int64 `json:"totalSeconds"`
+}
+
+// AudioMediaInfo is part of AudioTags.
+type AudioMediaInfo struct {
+	AudioFormat     string `json:"audioFormat"`
+	AudioBitrate    int64  `json:"audioBitrate"`
+	AudioChannels   int    `json:"audioChannels"`
+	AudioBits       int    `json:"audioBits"`
+	AudioSampleRate int    `json:"audioSampleRate"`
+}
+
+// AudioCountry is part of AudioTags.
+type AudioCountry struct {
+	TwoLetterCode string `json:"twoLetterCode"`
+	Name          string `json:"name"`
+}
+
+// ArtistTitleInfo is part of AudioTags.
+type ArtistTitleInfo struct {
+	Title            string `json:"title"`
+	TitleWithoutYear string `json:"titleWithoutYear"`
+	Year             int    `json:"year"`
+}
+
+// GetTrackFilesForArtist returns the track files for an artist.
+func (l *Lidarr) GetTrackFilesForArtist(artistID int64) ([]*TrackFile, error) {
+	return l.GetTrackFilesForArtistContext(context.Background(), artistID)
+}
+
+// GetTrackFilesForArtistContext returns the track files for an artist.
+func (l *Lidarr) GetTrackFilesForArtistContext(ctx context.Context, artistID int64) ([]*TrackFile, error) {
+	var output []*TrackFile
+
+	uri := bpTrackFile + "?artistId=" + strconv.FormatInt(artistID, starr.Base10)
+	if err := l.GetInto(ctx, uri, nil, &output); err != nil {
+		return nil, fmt.Errorf("api.Get(%s): %w", uri, err)
+	}
+
+	return output, nil
+}
+
+// GetTrackFilesForAlbum returns the track files for an album.
+func (l *Lidarr) GetTrackFilesForAlbum(albumID int64) ([]*TrackFile, error) {
+	return l.GetTrackFilesForAlbumContext(context.Background(), albumID)
+}
+
+// GetTrackFilesForAlbumContext returns the track files for an album.
+func (l *Lidarr) GetTrackFilesForAlbumContext(ctx context.Context, albumID int64) ([]*TrackFile, error) {
+	var output []*TrackFile
+
+	uri := bpTrackFile + "?albumId=" + strconv.FormatInt(albumID, starr.Base10)
+	if err := l.GetInto(ctx, uri, nil, &output); err != nil {
+		return nil, fmt.Errorf("api.Get(%s): %w", uri, err)
+	}
+
+	return output, nil
+}
+
+// GetTrackFiles returns the requested track files by ID.
+func (l *Lidarr) GetTrackFiles(trackFileIDs []int64) ([]*TrackFile, error) {
+	return l.GetTrackFilesContext(context.Background(), trackFileIDs)
+}
+
+// GetTrackFilesContext returns the requested track files by their IDs.
+func (l *Lidarr) GetTrackFilesContext(ctx context.Context, trackFileIDs []int64) ([]*TrackFile, error) {
+	var output []*TrackFile
+
+	if len(trackFileIDs) == 0 {
+		return output, nil
+	}
+
+	uri := bpTrackFile + "?"
+
+	for idx, fileID := range trackFileIDs {
+		if idx != 0 {
+			uri += "&"
+		}
+
+		uri += "trackFileIds=" + strconv.FormatInt(fileID, starr.Base10)
+	}
+
+	if err := l.GetInto(ctx, uri, nil, &output); err != nil {
+		return nil, fmt.Errorf("api.Get(%s): %w", uri, err)
+	}
```

Comment:
what do you think of using the `url` package instead? you already support query in `GetInto()` method.
something like this:
```go
query := url.Values{}
for _, fileID := range trackFileIDs {
	query.Add("trackFileIds",strconv.FormatInt(fileID, starr.Base10))
}

if err := l.GetInto(ctx, bpTrackFile, query, &output); err != nil {
	return nil, fmt.Errorf("api.Get(%s): %w", uri, err)
}
```
The `url` package has also a method for joining path `url.JoinPath()` that should behave the same as `path.Join()` if you're going to use `url`, maybe it's worth using this as well to avoid importing both.

Reply:
I don't know why I did it this way haha thanks